### PR TITLE
Disable triage-workflow staleness updates

### PR DESCRIPTION
**Pull request by eblume**
This commit changes the `triage-issues` workflow to no longer automatically remove the `stale` tag when updated. Instead, project managers with proper permissions will need to remove the label manually whenever that is deemed necessary.

This automation will reduce the number of volunteer contact points required to handle tickets that have not received attention from our more knowledgeable and code-contributing volunteers and ensure their time isn't wasted helping aspiring volunteers to figure out how to submit valid PRs to resolve tickets. This will keep the number of unresolved tickets low, which is of course the most important metric for an OSS community-driven project like ours.

Thanks to the project volunteer who, although unable to provide help for an obviously broken and easily fixed package[0] (#153199), was able to spare time to close my "really long stale" issue that "nobody has any idea how to fix". He helpfully pointed out that despite my included patch, the issue should have been "closed ages ago" because:

> There are no additional support channels. If no volunteer is
> interested in your issue then no volunteer time will be spend[sic] on
> it, it's that simple.

After meditating on this, I realized the wisdom of his words as they pointed out a fallacy: Indeed, time IS spent on every single issue that remains open, when the issue opener cares enough to keep bumping it. It's the time spent closing the issue, like he had done! So clever, and such a hackerly way to point that out to me.

In the same hacker spirit, I present this PR to fix the underlying issue. Alas I have also not tested this patch, this time not because of inscrutable build errors (I mean it's a missing system library, right?), but rather because I've decided to try to learn Nix again and I have to figure out how to get a flake to drop a service run script in the right directory, and it's 2024 and I'm in hell.

init save me.

~eblume

PS I am looking for BE/DevOps employment for fully remote positions.

[0]: https://github.com/Homebrew/homebrew-core/issues/153199

**This space left intentionally blank:**

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: .github/workflows/triage-issues.yml [hunk 1]
```diff
@@ -37,6 +37,7 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           days-before-stale: 21
           days-before-close: 7
+          ignore-updates: true
```
Comment (eblume) [inline .github/workflows/triage-issues.yml:40]: Reviewers please note: [ignore-issue-updates](https://github.com/actions/stale?tab=readme-ov-file#ignore-issue-updates) may be preferred.

**Comment by reitermarkus** [comment]
> an obviously broken and easily fixed package

> despite my included patch, the issue should have been "closed ages ago"

So can you submit that patch as a PR instead? That way it may actually be “easily fixed”.

Also note this PR won't do anything since the workflow is synced from https://github.com/Homebrew/.github/blob/master/.github/workflows/triage-issues.yml.

**Comment by p-linnane** [comment]
For a single line PR, there is quite a lot to unpack here.

> This commit changes the triage-issues workflow to no longer automatically remove the stale tag when updated. Instead, project managers with proper permissions will need to remove the label manually whenever that is deemed necessary.

The current behavior is sufficient. There is no issue with bumping an issue or PR a few times. We have a very limited amount of maintainers, and a lack of response may simply mean we have no answer for you. Given the volume of issues and PR's we receive, it may not be feasible to respond to every single one since this project is run entirely by volunteers.

> This automation will reduce the number of volunteer contact points required to handle tickets that have not received attention from our more knowledgeable and code-contributing volunteers and ensure their time isn't wasted helping aspiring volunteers to figure out how to submit valid PRs to resolve tickets.

I take issue with this line. It drips of sarcasm and entitlement, and is not representative of how we spend our time maintaining this project. It also shows a lack of understanding of the size and scale of this project. We are happy to review and provide suggestions for PR's. An issue for a single package that has had barely over 400 installs in the last year may be looked over while we prioritize other packages with greater impact.

> This will keep the number of unresolved tickets low, which is of course the most important metric for an OSS community-driven project like ours.

This is pretty far from the most important metric for a project like "ours". We package critical software such as Tor that allows individuals in countries with oppressive governments to freely utilize the Internet despite attempts to stop them. In my eyes, ensuring packages like that are working and available is well over 100x more important of a metric than a low issue count.

> Thanks to the project volunteer who, although unable to provide help for an obviously broken and easily fixed package https://github.com/Homebrew/homebrew-core/issues/153199, was able to spare time to close my "really long stale" issue that "nobody has any idea how to fix". He helpfully pointed out that despite my included patch, the issue should have been "closed ages ago" because:
>> There are no additional support channels. If no volunteer is
interested in your issue then no volunteer time will be spend[sic] on
it, it's that simple.

I'd like to give you the benefit of the doubt with how you worded this. As a community-driven project, we depend on users to contribute PR's. If the package you are interested in is "obviously broken and easily fixed" then the hope is that you will open a PR with a proposed fix. If you you are unable to successfully complete the PR, we will provide suggestions and help you.

> After meditating on this, I realized the wisdom of his words as they pointed out a fallacy: Indeed, time IS spent on every single issue that remains open, when the issue opener cares enough to keep bumping it. It's the time spent closing the issue, like he had done! So clever, and such a hackerly way to point that out to me.

This is completely unnecessary and rude. I'd like to point you to some basic reading that will help you better understand interacting with volunteers operating a massive project in their spare time: [Open Source Maintainers Owe You Nothing](https://mikemcquaid.com/open-source-maintainers-owe-you-nothing/). This is authored by our Project Leader @MikeMcQuaid who has been leading Homebrew for well over a decade and seen comments like yours countless times.

> In the same hacker spirit, I present this PR to fix the underlying issue. Alas I have also not tested this patch, this time not because of inscrutable build errors (I mean it's a missing system library, right?), but rather because I've decided to try to learn Nix again and I have to figure out how to get a flake to drop a service run script in the right directory, and it's 2024 and I'm in hell.

I wish you well with your attempts to learn Nix again. They operate a fantastic project and I hope it works better for you than Homebrew. If you are going to interact with the folks at Nix, I'd like to ask you to slightly scroll up and read the article I provided before you do so. If that's too difficult, here it is again: [Open Source Maintainers Owe You Nothing](https://mikemcquaid.com/open-source-maintainers-owe-you-nothing/).

> PS I am looking for BE/DevOps employment for fully remote positions.

Hands down the best comedy I've seen this year.

In conclusion, I do not expect you to respond appropriately to my writeup given your behavior thus far. PR's are very appreciated in this project, but this one was opened with a misguided and weak sense of malice that we are not interested in entertaining. I am happy to permanently ban you from interacting with Homebrew on GitHub if you find it necessary to try and prove some kind of point by responding to me. I'm too busy providing support to our many users who offer a basic level of respect and request help without snark. Thanks!

**Comment by eblume** [comment]
Hi Markus, thanks for responding. I’m not at my workstation so am unable to link you. Yes, if you click the link you will see this was covered. If you believe the correct action here is for me to publish the original PR without having tested it so that I can get help that way, I am happy to do so, but the multiple warnings not to do so held me back.As for the workflow syncing issue, I’ll look at that later today but I think anyone interested in the change can probably take it from here.On Feb 8, 2024, at 2:42 PM, Markus Reiter ***@***.***> wrote:﻿

an obviously broken and easily fixed package


despite my included patch, the issue should have been "closed ages ago"

So can you submit that patch as a PR instead? That way it may actually be “easily fixed”.
Also note this PR won't do anything since the workflow is synced from https://github.com/Homebrew/.github/blob/master/.github/workflows/triage-issues.yml.
